libraries/chain/fork_database.cpp

```diff
@@ -288,12 +245,7 @@ namespace eosio::chain {
          fc::raw::pack( out, *(*itr) );
       }
 
-      if( head ) {
-         fc::raw::pack( out, head->id() );
-      } else {
-         elog( "head not set in fork database; '${filename}' will be corrupted",
-               ("filename", fork_db_file) );
-      }
+      fc::raw::pack( out, head ? head->id() : digest_type());
```

Comment:
Seems we should fail and remove the file if no `head`. Writing a corrupt file is only setting up a time bomb.

Reply:
Actually I just tried again and skipping writing the file if no `head`  seems to work fine. Pushing this to the branch now.
I also added removing the file if we try to open it and can't find the head. 
Please let me know if this is OK or I should remove it.